docs/tutorial/fdfitting.rst

```diff
@@ -0,0 +1,343 @@
+
+
+Note how we added three models and then inverted the composition of those models. The parentheses 
+are important here, since otherwise we would have only inverted the offset model. Note that models
+inverted via `invert()` will typically be slower than the pre-inverted counterparts. This is because
+the inversion is done numerically, rather than analytically. For more complex examples on how this
+inversion may be used, please see the examples.
+
+For a full list of models that are available, please refer to the documentation by invoking
+`help(pylake.fitting.models)`.
+
+Fitting data
+------------
+
+To fit Fd models, we have to create an `FdFit`. This object will collect all the parameters
+involved in the models and data, and will allow you to interact with the model parameters
+and fit them. We construct it using `pylake.FdFit` and passing it one or more models. In
+return, we get an object we can interact with, which in this case we store in `odijk_fit`::
+
+    odijk_fit = pylake.FdFit(odijk_with_offsets)
+
+Adding data to the fit
+----------------------
+
+Next up, is adding some data. Let's assume we have two data sets. One was acquired in the presence
+of a ligand, and another was measured without a ligand. We expect this ligand to only affect the
+contour length of our DNA. Adding the first data set to the fit is simple::
+
+    odijk_fit.add_data("Control", force1, distance1)
+
+For the second data set, we want the contour length to be different. We can achieve this by renaming
+it when loading the data::
+
+    odijk_fit.add_data("RecA", force2, distance2, params={"DNA/Lc": "DNA/Lc_RecA"})
+
+More specifically, we renamed the parameter `DNA/Lc` to `DNA/Lc_RecA`. Sometimes, you may want
+a large number of data sets with different offsets. Assuming we have two lists of distance and
+force vectors stored in the lists distances and forces. In this case, it may make sense to load
+them in a loop and set such transformations programmatically::
+
+    for i, (distance, force) in enumerate(zip(distances, forces)):
+        odijk_fit.add_data(f"RecA {i}", f=force, d=distance, params={"DNA/f_offset": f"DNA/f_offset_{i}"})
+
+The syntax `f"DNA/f_offset_{i}"` is parsed into `DNA/f_offset_0`, `DNA/f_offset_1` ... etc. For more
+information on how this works, read up on Python fantastic f-Strings.
+
+Setting parameter bounds
+------------------------
+
+The parameters of the model can be accessed directly from `FdFit`. Note that by default,
+parameters tend to have reasonable initial guesses and bounds in Pylake, but we can set
+our initial guess and a lower and upper bound as follows::
+
+    odijk_fit["DNA/Lp"].value = 50
+    odijk_fit["DNA/Lp"].lower_bound = 39
+    odijk_fit["DNA/Lp"].upper_bound = 80
+
+After this, the model is ready to be fitted::
+
+    odijk_fit.fit()
+
+Note that multiple models can be fit at once, by just supplying more than one model::
```

Comment:
Ah yes, I should elaborate more. As far as I know, all functions of the `FdFit` object should still just work. You can just `plot` a `FdFit` that wasn't fitted for instance. It just fits the data poorly.  I do want to keep the ability to plot it before fitting as this can be useful for finding a good initial guess.

Since parameter estimation by convex optimization is often quite dependent on the initial guess, one might want to play with parameters first before actually fitting the model.

I changed the description to:
```rst
After this, the model is ready to be fitted. We can fit the model to the data by calling
the function `.fit()`. This estimates the model parameters by minimizing the least squares
differences between the model's dependent variable and the data in the fit::

    fit.fit()

After this call, the parameters will have new values that should bring the model closer
to the data. 
```

Reply:
Cool, sounds good. Maybe add a bit of the explanation from the comment as well. E.g. it's nice to know that you can plot before fitting and that there's a purpose for that.

Reply:
Alright. Added the following to the plotting section:

```rst
A model can be plotted before it is fitted. This can be useful when the default parameter
values don't seem to work very well. Parameter estimation is typically initiated from
an initial guess. A poor initial guess can lead to a poor parameter estimate. Therefore, 
you might want to see what your initial model curve looks like and set some better
initial guesses yourself when you run into trouble.
```

As it felt a bit more natural there. I'll have one more once-over before committing.